nixos/modules/services/networking/nm-file-secret-agent.nix

```diff
@@ -0,0 +1,135 @@
+{
+  config,
+  lib,
+  pkgs,
+  ...
+}:
+with lib;
+let
+  cfg = config.networking.networkmanager;
+  toml = pkgs.formats.toml { };
+
+  nmFileSecretAgentConfig = {
+    entry = builtins.map (
+      i:
+      {
+        key = i.key;
+        file = i.file;
+      }
+      // optionalAttrs (i.matchId != null) { match_id = i.matchId; }
+      // optionalAttrs (i.matchUuid != null) { match_uuid = i.matchUuid; }
+      // optionalAttrs (i.matchType != null) { match_type = i.matchType; }
+      // optionalAttrs (i.matchIface != null) { match_iface = i.matchIface; }
+      // optionalAttrs (i.matchSetting != null) {
+        match_setting = i.matchSetting;
+      }
+    ) cfg.ensureProfiles.secrets.entries;
+  };
+  nmFileSecretAgentConfigFile = toml.generate "config.toml" nmFileSecretAgentConfig;
+in
+{
+  meta = {
+    maintainers = [ maintainers.lilioid ];
+  };
+
+  ####### interface
+  options = {
+    networking.networkmanager.ensureProfiles.secrets = {
+      enable = mkOption {
```

Comment:
The original thought was so that a user could manually override it and enable the daemon while configuring it via another means. Now that you mention it, since there is no global configuration file which could be use for such a purpose, having the option to force enabelement does not make sense.